crates/worker/src/partition/mod.rs

```diff
@@ -530,6 +565,8 @@ where
                             &mut vqueues,
                         ).await?;
 
+                        self.update_last_applied_log_lsn(lsn, &started_at);
```

Comment:
Why is it important to update the last applied log lsn after applying the record and not keeping it within the `maybe_advance` method? Now, we have two places where we need to call it.